gtsfm/frontend/verifier/loransac.py

```diff
@@ -0,0 +1,194 @@
+"""
+Wrapper about COLMAP's LORANSAC Essential matrix estimation, using pycolmap's pybind API.
+
+LORANSAC paper:
+ftp://cmp.felk.cvut.cz/pub/cmp/articles/matas/chum-dagm03.pdf
+
+On Linux, a python wheel is available:
+(add URL)
+
+On Mac, no wheel is available, so system-wide installation of COLMAP is required using this module.
+   Follow the instructions first here: https://colmap.github.io/install.html
+   and then here: https://github.com/mihaidusmanu/pycolmap#getting-started
+
+Authors: John Lambert
+"""
+
+from typing import Any, Dict, Optional, Tuple
+
+import numpy as np
+
+import pycolmap
+from gtsam import Cal3Bundler, Rot3, Unit3
+
+import gtsfm.utils.logger as logger_utils
+import gtsfm.utils.verification as verification_utils
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.verifier.verifier_base import VerifierBase, NUM_MATCHES_REQ_E_MATRIX, NUM_MATCHES_REQ_F_MATRIX
+
+
+logger = logger_utils.get_logger()
+
+
+class LoRansac(VerifierBase):
+    def __init__(
+        self,
+        use_intrinsics_in_verification: bool,
+        estimation_threshold_px: float,
+        min_allowed_inlier_ratio_est_model: float,
+    ) -> None:
+        """Initializes the verifier.
+
+        Note: LoRANSAC is hard-coded in pycolmap to use the following hyperparameters:
+            min_inlier_ratio = 0.01
+            min_num_trials = 1000
+            max_num_trials = 100000
+            confidence = 0.9999
+
+        (See https://github.com/mihaidusmanu/pycolmap/blob/master/essential_matrix.cc#L98)
+
+        Args:
+            use_intrinsics_in_verification: Flag to perform keypoint normalization and compute the essential matrix
+                instead of fundamental matrix. This should be preferred when the exact intrinsics are known as opposed
```

Comment:
is this indentation added by subblack?

Reply:
no, it was just something i added. but i see indentation here too: https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

Reply:
Its nice to have, but it's hard to maintain consistency across codebase, so I would not prefer it. Upto you.